### DISCUSSION TITLE
ACAP 3.1

### DISCUSSION THREAD

**Discussion by CCho4** [AxisCommunications/acap3-examples#208] [Q&A] [answered]
I'm using a M3046-V and 9.8 latest firmware for it. So SDK 3.1 is the latest I can use. I've downloaded the 3.1-3.+ SDKs but they only compile with 3.5 and I they won't upload to my cameras. If I change them to 3.1 in the docker file they fail. Any ideas PLEASE and thanks! (even hello world won't compile to 3.1...)

**Comment by pataxis** [comment, marked as answer]
Hi @CCho4 , thanks for your question.

For products running AXIS OS 9.80 it's required to build applications with a [compatible SDK](https://help.axis.com/en-us/acap-3-developer-guide#find-the-right-sdk-for-software-compatibility), in this case ACAP SDK 3.1 as you have tried.
To ensure examples that works for the SDK you want to use, you also need to checkout tag [3.1](https://github.com/AxisCommunications/acap3-examples/tree/3.1) of this repository.

In short, start out by checking out tag 3.1 of this repo and try to build the hello-world example which specifies the 3.1 SDK in the Dockerfile.
